railties/lib/rails/commands/test/test_command.rb

```diff
@@ -69,7 +69,8 @@ def generators(*args)
       end
 
       private
-        EXACT_TEST_ARGUMENT_PATTERN = /^-n|^--name\b|#{Rails::TestUnit::Runner::PATH_ARGUMENT_PATTERN}/
+        EXACT_TEST_ARGUMENT_PATTERN = /^(-n\b|--name\b|[^-])/
+        private_constant :EXACT_TEST_ARGUMENT_PATTERN
```

Comment:
Two things:
1. `EXACT_TEST_ARGUMENT_PATTERN` is meant to match arguments which are any of those three:
    1. `-n`
    2. `--name`
    3. Any test file path.
      This one used to be based on `PATH_ARGUMENT_PATTERN` which was part of the problem this PR is solving.
      Realistically, any argument that does not start with a hyphen `-` _can be_ a test file path argument.
      In practice, it might not always be: it could be for example a seed passed to `--seed` or an exclude path passed to `--exclude` but since the code in Rails seems to [avoid using `OptionParser` to parse all Minitest options](https://github.com/rails/rails/blob/d444e7ff8766f67dbfe953d7eb9058e1cb0bf6c9/railties/lib/rails/test_unit/runner.rb#L32) (too tight coupling?), this seemed to be the easy way out.
      If there was a way to reuse Minitest's option parser, we could do better.
2. I might as well mark the constant as private, since it was meant to be.

Reply:
Turns out I was wrong, this is not an easy way out!
```
Error:
Rails::Command::TestTest#test_test_command_with_options_runs_test:prepare_task:
RuntimeError: rails command failed (1): bin/rails test --seed 1234 -e development 2>&1
/usr/local/lib/ruby/3.3.0/bundled_gems.rb:74:in `require': cannot load such file -- /rails/tmp/d20240623-1161-xi2hba/app/1234 (LoadError)
    from /usr/local/lib/ruby/3.3.0/bundled_gems.rb:74:in `block (2 levels) in replace_require'
```